src/Engine/ProtoCore/DSASM/Heap.cs

```diff
@@ -308,7 +318,7 @@ public Heap()
         /// <param name="values">Array elements whose indices are integer</param>
         /// <param name="dict">Array elements whose indices are not integer</param>
         /// <returns></returns>
-        public StackValue AllocateArray(IEnumerable<StackValue> values)
+        public StackValue AllocateArray(StackValue[] values)
```

Comment:
This is good, I'm just curious about the intent. Do we make it an array because the name says `AllocateArray`? Does it cause callers to call `ToArray` now that they did have to do in the past? Just want to understand the rationale.